js/utils.js

```diff
@@ -0,0 +1,45 @@
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const data = await response.json();
+    const status = response.status;
+    const res = {
+      data,
+      status,
+    };
+    return res;
+  } catch (err) {
+    throw err;
+  }
+}
+
+function createElement({ type, classList = [], id }) {
+  const element = document.createElement(type);
+  element.classList.add(...classList);
+  element.id = id ?? isEmpty(id);
+  return element;
+}
+
+/**
+ * @param value:
+ * @returns boolean which returns
+ * - `true` if value is empty or falsy
+ */
+function isEmpty(valueToCheck) {
+  if (typeof valueToCheck === 'undefined') {
```

Comment:
is this function complete?

Reply:
> Make all the `png `into `svg`. Please format the code. It's difficult to read the logic inside the `generateSavedDetailsForm`. Attaching a SS below as an example of ![image](https://user-images.githubusercontent.com/64726586/229686678-cefd8b3c-efb7-4cc3-a5b6-8afc17b12ffa.png)

Hello vaibhav 
I am using two image as png in my particular file i.e. img/intro-page/page-not-authorized.png and page-not-found.png

and others image is globally used in all files including mine in navbar. So my question is should i convert png to svg only image of intro page ??

Reply:
make all your pngs into svgs

Reply:
also @shubhamsigdar1 what is this `isEmpty ` function going to check?  The code still looks the same to me

Reply:
idEmpty function is checking Is `id` of element is undefine or null or not
if it is undefine then give `id=true` but i think i have to remove this function and do something like this...If I am wrong please correct me
```function createElement({ type, classList = [], id }) {
  const element = document.createElement(type);
  element.classList.add(...classList);
  element.id = id ?? true;
  return element;
}```